### ISSUE TITLE
Feature request: deeply merge_configs

### ISSUE THREAD

**Issue by Jasha10** [dbatten5/maison#88]
Using the new `merge_configs` option to the `ProjectConfig.__init__` method, configs like the following can be merged:
```
merged({"foo": "bar"}, {"baz": "qux"}) == {"foo": "bar", "baz": "qux"})
```

However, dictionaries nested more deeply than the toplevel are not merged:
```
# current behavior:
merged({"inner": {"foo": "bar"}}, {"inner": {"baz": "qux"}}) == {"inner": {"baz": "qux"}})
# left config overwritten by right config
```
I would expect that nested configuration gets merged instead of being overwritten:
```
# expected:
merged({"inner": {"foo": "bar"}}, {"inner": {"baz": "qux"}}) == {"inner": {"foo": "bar", "baz": "qux"}})
```

**Comment by Jasha10** [comment]
For context: deep merging will be useful for [autoimport](https://github.com/lyz-code/autoimport). Specifically, deep merging will enable the use-case where multiple autoimport config files each specify part of the [`common_statements`](https://lyz-code.github.io/autoimport/#configuration) dict.

cc @lyz-code

**Comment by dbatten5** [comment]
hi @Jasha10 thanks for raising the issue and the pr, i like the proposal. will take a look at the pr now

**Comment by dbatten5** [comment]
@Jasha10 @lyz-code this should now be available in 1.4.0
